### DISCUSSION TITLE
how to consume a GraphQL endpoint?

### DISCUSSION THREAD

**Discussion by Cerchie** [thomasheyenbrock/remix-graphql#7] [General]
To replicate the issue I'm running into, on MacOS, I ran:
```
~ % npx create-remix@latest
Need to install the following packages:
  create-remix@latest
Ok to proceed? (y) y
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
? Where would you like to create your app? ./remix-experiment
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure; it's easy to 
change deployment targets. Remix App Server
? Do you want me to run `npm install`? Yes
```

I then ran `npm run dev` and the app worked. 

The next thing I did was add a file called `$username.tsx` to the `/routes` folder, and copy-pasted the code for consuming graphql from the readme in this repository. I got this error: 
```
 File changed: app/routes/$username.tsx
💿 Rebuilding...
💿 Rebuilt in 68ms
Error: Did you forget to run `remix setup` for your platform?
    at Object.<anonymous> (/Users/luciacerchie/remix-experiment/node_modules/remix/index.js:22:7)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/luciacerchie/remix-experiment/node_modules/remix-graphql/dist/index.js:35:21)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
GET / 500 - - 68.833 ms
^C
```
It's odd because my repository doesn't have that script as generated.

I'm wondering what part of the remix architecture I could be missing to make this work? Thanks in advance for any help!

**Comment by thomasheyenbrock** [comment]
Hi @Cerchie 👋  just tried this out and can reproduce this. Could you please open an issue for this as this is likely a bug of this package that needs fixing: https://github.com/thomasheyenbrock/remix-graphql/issues/new/choose

**Reply by Cerchie** [reply]
[Of course! ](https://github.com/thomasheyenbrock/remix-graphql/issues/8)

I'd love to help out with it in whatever way I can as well. 

Meanwhile, do you mind if I make a quick PR to adjust a code block in the README?

I noticed you've got the old way of importing remix in it:

<img width="769" alt="Screen Shot 2022-06-14 at 6 27 29 AM" src="https://user-images.githubusercontent.com/54046179/173589360-9cf9a442-1c10-4404-aa82-14379c4cdcaa.png">

It should be `import type { LoaderFunction } from "@remix-run/node"; `

**Reply by thomasheyenbrock** [reply]
Feel free to open a PR for that 🙌
